sim/team-validator.ts

```diff
@@ -200,14 +200,22 @@ export class TeamValidator {
 			this.ruleTable.minSourceGen[0] : 1;
 	}
 
-	validateTeam(team: PokemonSet[] | null, removeNicknames: boolean = false): string[] | null {
+	validateTeam(
+		team: PokemonSet[] | null,
+		removeNicknames = false,
```

Comment:
With 2 optionals its less obvious whether its necessary or not, but this could potentially be changed into a `{removeNicknames?: boolean, skipSets?: Record<string, Record<string, boolean>>}` option type if we saw more optionals being added. I think its OK for now, but would be happy to change it

Reply:
What's a Record?

Reply:
I think this signature should be refactored to an options object.

Reply:
`Record<string, number>` => `{[key: string]: number}`

Reply:
> I think this signature should be refactored to an options object.

Done (but leaving this thread unresolved in case you were curious about `Record`. I switched away from it, I don't think it adds value here it was just easier to type originally)

Reply:
Okay, I understand what a Record is now, but `Record<string, boolean>` seems less readable than `{[k: string]: boolean}` to me.

Reply:
Agreed, which is why I switched back

Reply:
I think it's even less readable several levels deep, and an `interface` with proper indentation would be better in that case.